src/components/PeopleContext/PeopleContext.tsx

```diff
@@ -0,0 +1,88 @@
+import React, {
+  createContext, useContext, useEffect, useState, ReactNode,
+} from 'react';
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+
+interface PeopleContextProps {
+  loading: boolean;
+  people: Person[];
+
+  searchByName:string;
+  setSearchByName: React.Dispatch<React.SetStateAction<string>>;
+
+  filteredPeople: Person[];
+  setFilteredPeople: React.Dispatch<React.SetStateAction<Person[]>>;
+
+  errorMessage: string | null;
+  isMotherInArray: (name: string | null) => boolean;
+  isFatherInArray: (name: string | null) => boolean;
+}
+
+const PeopleContext = createContext<PeopleContextProps | undefined>(undefined);
+
+interface PeopleProviderProps {
+  children: ReactNode;
+}
+
+export const PeopleProvider: React.FC<PeopleProviderProps> = ({ children }) => {
+  const [loading, setLoading] = useState(true);
```

Comment:
Rename all boolean variables using this [tips](https://dev.to/michi/tips-on-naming-boolean-variables-cleaner-code-35ig)

```suggestion
  const [isLoading, setIsLoading] = useState(true);
```